### PR TITLE
otelcol.exporter.file: add file exporter component with configu…

### LINKED ISSUE #4398: Include Community-Supported fileexporter in Grafana Alloy
### Community component name

fileexporter
Details  of the component is here 
https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/fileexporter
Its also included in all otel distribution '[core](https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol), [contrib](https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib), [k8s](https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-k8s)'

[More details of the use case are included here] (https://github.com/grafana/alloy/issues/4262#issuecomment-3280373643) 

### Community component maintainer

- [ ] I agree to be a maintainer for this component.

### Community component description

Writes telemetry data to files on disk.
Exporter supports the following features：

- Support for writing pipeline data to a file.
- Support for rotation of telemetry files.
- Support for compressing the telemetry data before exporting.
- Support for writing into multiple files, where the file path is determined by a resource attribute.

[More details of the use case are included here] (https://github.com/grafana/alloy/issues/4262#issuecomment-3280373643) 

### Tip

<sub>React with 👍 if this issue is important to you.</sub>

### PR DESCRIPTION
**Pull request by madhub**
This pull request introduces a new `otelcol.exporter.file` component, which allows metrics, logs, and traces to be written to disk with options for rotation, compression, and grouping by resource attribute. The implementation includes documentation, code, dependency updates, and thorough validation and testing. This addition enables users to persist telemetry data locally in either JSON or Protocol Buffers format and provides flexible configuration for file management.

**New Feature: File Exporter Component**

* Added the `otelcol.exporter.file` component, enabling telemetry data export to disk with support for file rotation, compression (zstd), and grouping output files by resource attribute. The component wraps the upstream OpenTelemetry Collector Contrib `fileexporter`. [[1]](diffhunk://#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR15) [[2]](diffhunk://#diff-62447074a850020f57d18f1aa453d477f6cbbe5ef12700e844b49f4de1903a74R1-R222)

**Documentation and Usage**

* Created comprehensive documentation for `otelcol.exporter.file`, detailing arguments, blocks (rotation, group_by, debug_metrics), configuration options, usage examples, and technical details.

**Codebase Integration**

* Registered the new component in the Alloy system and ensured it is available for use by importing it in the component registry (`all.go`).

**Dependency Management**

* Added `github.com/open-telemetry/opentelemetry-collector-contrib/exporter/fileexporter` as a dependency in `go.mod` to support the new exporter.

**Validation and Testing**

* Implemented robust validation logic for configuration options, including edge cases for path, format, compression, rotation, and group_by settings. Added unit tests to verify defaulting, validation, conversion, and unmarshalling of configuration.

This change provides a flexible and robust mechanism for exporting telemetry data to files, suitable for local debugging, archival, or integration with other file-based workflows.…ration options

#### Which issue(s) this PR fixes

Fixes #4398

#### Notes to the Reviewer
Following tests were done
1. Formatting test with file exporter
	`./alloy fmt example-file-exporter.alloy`
2. Run test with file OTLP input & file export
	`./alloy run test-file-exporter.alloy`
3. Conversion test OpenTelemetry configuration and converting it to alloy
	`./alloy run test-file-exporter.alloy`

4. End 2 end test : OTLP input , file export & sending sample OTel Log record using Curl 
	`./alloy run test-converted.alloy`

**Alloy configuration used for testing**
```text
// Example configuration showing otelcol.exporter.file working with other components

// A test receiver that generates traces
otelcol.receiver.otlp "test" {
  grpc {
    endpoint = "127.0.0.1:4317"
  }
  
  http {
    endpoint = "127.0.0.1:4318"
  }
  
  output {
    metrics = [otelcol.exporter.file.json_output.input]
    logs    = [otelcol.exporter.file.json_output.input] 
    traces  = [otelcol.exporter.file.json_output.input]
  }
}

// File exporter with JSON format
otelcol.exporter.file "json_output" {
  path   = "/tmp/alloy-telemetry.json"
  format = "json"
  flush_interval = "2s"
  
  debug_metrics {
    disable_high_cardinality_metrics = true
  }
}

// File exporter with rotation
otelcol.exporter.file "rotated_output" {
  path   = "/tmp/alloy-rotated.json"
  format = "json"
  
  rotation {
    max_megabytes = 10
    max_days      = 7
    max_backups   = 5
    localtime     = true
  }
  
  debug_metrics {
    disable_high_cardinality_metrics = true
  }
}

// File exporter with compression
otelcol.exporter.file "compressed_output" {
  path        = "/tmp/alloy-compressed.json"
  format      = "proto"
  compression = "zstd"
  
  debug_metrics {
    disable_high_cardinality_metrics = true
  }
}

// Note: Debug exporter removed as it requires --stability.level=experimental
```
**Example test message sent via curl**
```bash
curl -X POST http://localhost:4318/v1/logs   -H "Content-Type: application/json"   -d '{
    "resourceLogs": [
      {
        "resource": {
          "attributes": [
            { "key": "service.name", "value": { "stringValue": "demo-service" } },
            { "key": "service.version", "value": { "stringValue": "1.0.0" } },
            { "key": "host.name", "value": { "stringValue": "demo-host" } }
          ]
        },
        "scopeLogs": [
          {
            "scope": {
              "name": "demo-logger",
              "version": "1.0.0"
            },
            "logRecords": [
              {
                "timeUnixNano": "1694700000000000000",
                "severityText": "INFO",
                "body": { "stringValue": "Hello from OpenTelemetry logs!" },
                "attributes": [
                  { "key": "env", "value": { "stringValue": "dev" } },
                  { "key": "region", "value": { "stringValue": "ap-south-1" } }
                ]
              }
            ]
          }
        ]
      }
    ]
  }'
```

#### PR Checklist

- [x ] CHANGELOG.md updated
- [ x] Documentation added
- [x ] Tests updated
- [x ] Config converters updated

### CODE REVIEW

**Comment by madhub** [comment]
> [CLA assistant check](https://cla-assistant.io/grafana/alloy?pullRequest=4475) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/alloy?pullRequest=4475) before we can accept your contribution.
> **madhub** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/alloy?pullRequest=4475) it.

Done
image

**Comment by clayton-cornell** [comment]
There are some changes we need to make to the docs, but.. they are a lot to put into the GitHub webUI. Would you mind if I edited the doc input in your fork and pushed back to GitHub?

**Comment by madhub** [comment]
> There are some changes we need to make to the docs, but.. they are a lot to put into the GitHub webUI. Would you mind if I edited the doc input in your fork and pushed back to GitHub?

Pls go-ahead

**Comment by clayton-cornell** [comment]
@madhub I've made some changes to the doc and pushed to your branch. The changes are primarily style and linting. I reorganized a few bits to conform a bit better to the general style currently in-use in the component docs.  I also added the metadata and othe rinfo for Community written/supported components.

**Review by madhub — COMMENTED** [review]
ok

**Comment by madhub** [comment]
@clayton-cornell anything pending from my side ?,

**Comment by clayton-cornell** [comment]
Next is a code review from @grafana/grafana-alloy-maintainers

**Comment by drconopoima** [comment]
Is this extension with the "compression: zstd" feature enabled not adding a filename extension?

I see the testing code includes cases with compression and filename ".json"

```alloy
otelcol.exporter.file "compressed" {
  path        = "/tmp/traces.jsonl"
  format      = "proto"
  compression = "zstd"
}
```

```go
alloyCfg := `
path = "/tmp/*/test.json"
format = "json"
append = false
compression = "zstd"
flush_interval = "5s"
rotation {
  max_megabytes = 50
  max_days = 7
  max_backups = 10
  localtime = true
}
group_by {
  enabled = true
  resource_attribute = "service.name"
  max_open_files = 50
}
```

I searched for ".zst" or ".gz" and found no references in the code

**Comment by madhub** [comment]
> Is this extension with the "compression: zstd" feature enabled not adding a filename extension?
> 
> I see the testing code includes cases with compression and filename ".json"
> 
> ```alloy
> otelcol.exporter.file "compressed" {
>   path        = "/tmp/traces.jsonl"
>   format      = "proto"
>   compression = "zstd"
> }
> ```
> 
> ```go
> alloyCfg := `
> path = "/tmp/*/test.json"
> format = "json"
> append = false
> compression = "zstd"
> flush_interval = "5s"
> rotation {
>   max_megabytes = 50
>   max_days = 7
>   max_backups = 10
>   localtime = true
> }
> group_by {
>   enabled = true
>   resource_attribute = "service.name"
>   max_open_files = 50
> }
> ```
> 
> I searched for ".zst" or ".gz" and found no references in the code
Compression support is included the fileexporter, see the code here [fileexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ce1bb622659c2a7b1782a154f2d7d9dc0a9b81e2/exporter/fileexporter/codec.go#L25)
also here https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/ce1bb622659c2a7b1782a154f2d7d9dc0a9b81e2/exporter/fileexporter/factory.go#L38

**Comment by drconopoima** [comment]
I see in the repo you linked that for other contrib exporter and other compression algorithm it appended ".gz" to the files without specifying:
[as logs_103016847.json.gz, I set compression: gzip, but did not specify the file extension](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/37917)
I'm guessing the same behaviour applies to Zstd, but I was unable to find examples if it appended ".zst" or ".gz", as I have learned that Zstd is gzip compatible

**Comment by madhub** [comment]
> I see in the repo you linked that for other contrib exporter and other compression algorithm it appended ".gz" to the files without specifying: [as logs_103016847.json.gz, I set compression: gzip, but did not specify the file extension](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/37917) I'm guessing the same behaviour applies to Zstd, but I was unable to find examples if it appended ".zst" or ".gz", as I have learned that Zstd is gzip compatible


I tested it with zstd, and it seems the file is Zstd-compressed with a length prefix. This behavior is also documented here https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/fileexporter/README.md#file-format
image


Here is the sample output after removing the 4 bytes. You can clearly see zstd magic string `28b5 2ffd`

image

For Reference 
- [alloy-compressed.json](https://github.com/user-attachments/files/22645029/alloy-compressed.json)

- alloy configuration file with `otelcol.exporter.file.compressed_output.input` 
```text
// Example configuration showing otelcol.exporter.file working with other components

// A test receiver that generates traces
otelcol.receiver.otlp "test" {
  grpc {
    endpoint = "127.0.0.1:4317"
  }
  
  http {
    endpoint = "127.0.0.1:4318"
  }
  
  output {
    metrics = [otelcol.exporter.file.compressed_output.input]
    logs    = [otelcol.exporter.file.compressed_output.input] 
    traces  = [otelcol.exporter.file.compressed_output.input]
  }
}

// File exporter with JSON format
otelcol.exporter.file "json_output" {
  path   = "/tmp/alloy-telemetry.json"
  format = "json"
  flush_interval = "2s"
  
  debug_metrics {
    disable_high_cardinality_metrics = true
  }
}

// File exporter with rotation
otelcol.exporter.file "rotated_output" {
  path   = "/tmp/alloy-rotated.json"
  format = "json"
  
  rotation {
    max_megabytes = 10
    max_days      = 7
    max_backups   = 5
    localtime     = true
  }
  
  debug_metrics {
    disable_high_cardinality_metrics = true
  }
}

// File exporter with compression
otelcol.exporter.file "compressed_output" {
  path        = "/tmp/alloy-compressed.json"
  format      = "json"
  compression = "zstd"
  
  debug_metrics {
    disable_high_cardinality_metrics = true
  }
}

// Note: Debug exporter removed as it requires --stability.level=experimental

```

**Comment by drconopoima** [comment]
Thanks for the demonstration. It seems to have produced output in some format that could be characterized by file extension labels `.json.zst.bytecount`, but we'll have to instruct in configuration to add the extension labels.

Unrelated to this MR, just as an aside, it seems to have added a byte count even when using json format, while documented behaviour appears to indicate that it's only added for protocol buffers

**Comment by madhub** [comment]
> Thanks for the demonstration. It seems to have produced output in some format that could be characterized by file extension labels `.json.zst.bytecount`, but we'll have to instruct in configuration to add the extension labels.
> 
> Unrelated to this MR, just as an aside, it seems to have added a byte count even when using json format, while documented behaviour appears to indicate that it's only added for protocol buffers

I also tried setting the format to `proto` instead of `json`. The file is zstd-compressed with a **length prefix**. See the attached image
```text

// File exporter with compression
otelcol.exporter.file "compressed_output" {
  path        = "/tmp/alloy-compressed.proto"
  format      = "proto"
  compression = "zstd"

  debug_metrics {
    disable_high_cardinality_metrics = true
  }
}
```

image
@drconopoima  anything pending from my side ?

**Comment by drconopoima** [comment]
> @drconopoima anything pending from my side ?

My doubts are resolved, I'm not an approver/reviewer. In my view, the result of the compression is as they are coded for in the dependencies opentelemetry-contrib, the problem I see, in general the file type prefixed by length is not adequately labeled by a file extension.

From this point onwards I'm not talking about the merge, but the behaviour of added dependencies as per your helpful demonstrations:

If/and/when this is used, if the users aren't aware of the behaviour, many backed-up outputs may get discarded as unprocessable/corrupted if they don't strip out the length prefix (even in json, where the readme says it shouldn't appear), or aren't configuring alloy to append the compression file extension '.zst' suffix.

There is also no hashing to ensure that the length prefix was calculated for the right ensuing payload contents, it would be rather easy to zero-out and re-write the prefix with any other amount that ensures allocated memory will not suffice.

I think the general approach could be to store the hash of the length size alongside the payload, and prefix with the hash of both, then calculate the hash before trusting the length, to ensure it's not tampered with, otherwise consider the length prefix as not being present. Only issue is that this nullifies the desired effect of not needing to load the full file before allocating.

**Comment by madhub** [comment]
> > @drconopoima anything pending from my side ?
> 
> My doubts are resolved, I'm not an approver/reviewer. In my view, the result of the compression is as they are coded for in the dependencies opentelemetry-contrib, the problem I see, in general the file type prefixed by length is not adequately labeled by a file extension.
> 
> From this point onwards I'm not talking about the merge, but the behaviour of added dependencies as per your helpful demonstrations:
> 
> If/and/when this is used, if the users aren't aware of the behaviour, many backed-up outputs may get discarded as unprocessable/corrupted if they don't strip out the length prefix (even in json, where the readme says it shouldn't appear), or aren't configuring alloy to append the compression file extension '.zst' suffix.
> 
> There is also no hashing to ensure that the length prefix was calculated for the right ensuing payload contents, it would be rather easy to zero-out and re-write the prefix with any other amount that ensures allocated memory will not suffice.
> 
> I think the general approach could be to store the hash of the length size alongside the payload, and prefix with the hash of both, then calculate the hash before trusting the length, to ensure it's not tampered with, otherwise consider the length prefix as not being present. Only issue is that this nullifies the desired effect of not needing to load the full file before allocating.

since it just wrapper over [open-telemetry fileexporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/fileexporter/README.md#file-format) any enhancement should be raised on repo [opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib)

**Comment by madhub** [comment]
Hello everyone,
This pull request has been awaiting review and approval for about a month. Could we please get an update on the expected timeline for either reviewing and approving or rejecting it?

File: internal/component/otelcol/exporter/file/file.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+		return errors.New("compression must be zstd if specified")
+	}
+
+	if args.FlushInterval <= 0 {
+		return errors.New("flush_interval must be greater than zero")
+	}
+
+	if args.GroupBy != nil && args.GroupBy.Enabled {
+		// Apply implicit defaults if user enabled group_by but omitted optional fields.
+		if args.GroupBy.ResourceAttribute == "" {
```
Comment (dehaansa) [inline internal/component/otelcol/exporter/file/file.go]: Defaults set in this way won't display in the Alloy UI, can you set the defaults in the SetToDefault function?

File: internal/component/otelcol/exporter/file/file.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+}
+
+// DebugMetricsConfig implements exporter.Arguments.
+func (args Arguments) DebugMetricsConfig() otelcolCfg.DebugMetricsArguments {
+	return args.DebugMetrics
+}
+
+// Validate implements syntax.Validator.
+func (args *Arguments) Validate() error {
+	if args.Path == "" {
```
Comment (dehaansa) [inline internal/component/otelcol/exporter/file/file.go]: As the majority of this appears to be copied from upstream, I think it would be easier to maintain if we just call Convert() and then Validate() on the upstream struct.

File: internal/component/otelcol/exporter/file/file.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+	"github.com/grafana/alloy/internal/featuregate"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pipeline"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/fileexporter"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "otelcol.exporter.file",
+		Stability: featuregate.StabilityGenerallyAvailable,
```
Comment (dehaansa) [inline internal/component/otelcol/exporter/file/file.go]: As this is Alpha upstream, we would typically do first release as PublicPreview.

File: docs/sources/reference/components/otelcol/otelcol.exporter.file.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
+---
+canonical: https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.exporter.file/
+description: Learn about otelcol.exporter.file
+labels:
+  stage: general-availability
+  products:
+    - oss
+  tags:
+    - text: Community
+      tooltip: This component is developed, maintained, and supported by the Alloy user community.
```
Comment (dehaansa) [inline docs/sources/reference/components/otelcol/otelcol.exporter.file.md]:
```suggestion
```

I don't think this needs to be a community component, as it doesn't require non-grafana vendor environments to validate/test.

**Review by dehaansa — COMMENTED** [review]
Sorry for the delay on reviewing this! Overall looks good, had some minor feedback.

File: internal/component/otelcol/exporter/file/file.go (hunk 1 above)
Comment (madhub) [reply internal/component/otelcol/exporter/file/file.go]: done

File: internal/component/otelcol/exporter/file/file.go (hunk 3 above)
Comment (madhub) [reply internal/component/otelcol/exporter/file/file.go]: done

File: internal/component/otelcol/exporter/file/file.go (hunk 2 above)
Comment (madhub) [reply internal/component/otelcol/exporter/file/file.go]: done

File: internal/component/database_observability/postgres/collector/explain_plan.go [hunk 5]
```diff
@@ -151,3 +194,350 @@ func (p *PlanNode) totalCost() *float64 {
...
+		selectStatement = fmt.Sprintf(selectQueriesForExplainPlanTemplate, "s.stats_since")
+	} else {
+		statReset := c.dbConnection.QueryRowContext(ctx, "SELECT stats_reset FROM pg_stat_statements_info")
+		if err := statReset.Err(); err != nil {
+			return fmt.Errorf("failed to fetch stats reset time for explain plans: %w", err)
+		}
+		if err := statReset.Scan(&resetTS); err != nil {
+			return fmt.Errorf("failed to scan stats reset time for explain plans: %w", err)
+		}
+		selectStatement = fmt.Sprintf(selectQueriesForExplainPlanTemplate, "NOW() AT TIME ZONE 'UTC' AS stats_since")
```
Comment (ghost) [inline internal/component/database_observability/postgres/collector/explain_plan.go:354]: **Semgrep identified an issue in your code:**
String-formatted SQL query detected. This could lead to SQL injection if the string is not sanitized properly. Audit this call to ensure the SQL is not manipulable by external data.

**To resolve this comment:**



🔧 No guidance has been designated for this issue. Fix according to your organization's approved methods.

<details><summary>💬 Ignore this finding</summary>



Reply with Semgrep commands to ignore this finding.
- `/fp <comment>` for false positive
- `/ar <comment>` for acceptable risk
- `/other <comment>` for all other reasons


Alternatively, triage in [Semgrep AppSec Platform](https://semgrep.dev/orgs/grafana_labs/findings/297856440) to ignore the finding created by [string-formatted-query](https://semgrep.dev/playground/r/6xTxjW8/go.lang.security.audit.database.string-formatted-query.string-formatted-query?utm_campaign=finding_notification&utm_medium=review_comment&utm_source=github&utm_content=rule).


</details>

We're currently testing semgrep's diff-aware PR comment feature on a subset of our repos-- if you run into issues or find this spammy, please reach out to @danny.cooper in slack and give feedback. 

For backwards compatability with gosec, its best to use polyglot suppression comments of the following format for false positives:
`// #nosec <gosec rule ID> nosemgrep: <semgrep rule ID>`


<sub>*You can view more details about [this finding](https://semgrep.dev/orgs/grafana_labs/findings/297856440) in the Semgrep AppSec Platform.*</sub>

File: internal/component/database_observability/postgres/collector/explain_plan.go [hunk 6]
```diff
@@ -151,3 +194,350 @@ func (p *PlanNode) totalCost() *float64 {
...
+func (c *ExplainPlan) Stop() {
+	c.cancel()
+}
+
+func (c *ExplainPlan) populateQueryCache(ctx context.Context) error {
+	var selectStatement string
+	var resetTS time.Time
+	version17Plus := semver.MustParseRange(">=17.0.0")(c.dbVersion)
+	if version17Plus {
+		selectStatement = fmt.Sprintf(selectQueriesForExplainPlanTemplate, "s.stats_since")
```
Comment (ghost) [inline internal/component/database_observability/postgres/collector/explain_plan.go:345]: **Semgrep identified an issue in your code:**
String-formatted SQL query detected. This could lead to SQL injection if the string is not sanitized properly. Audit this call to ensure the SQL is not manipulable by external data.

**To resolve this comment:**



🔧 No guidance has been designated for this issue. Fix according to your organization's approved methods.

<details><summary>💬 Ignore this finding</summary>



Reply with Semgrep commands to ignore this finding.
- `/fp <comment>` for false positive
- `/ar <comment>` for acceptable risk
- `/other <comment>` for all other reasons


Alternatively, triage in [Semgrep AppSec Platform](https://semgrep.dev/orgs/grafana_labs/findings/297856441) to ignore the finding created by [string-formatted-query](https://semgrep.dev/playground/r/6xTxjW8/go.lang.security.audit.database.string-formatted-query.string-formatted-query?utm_campaign=finding_notification&utm_medium=review_comment&utm_source=github&utm_content=rule).


</details>

We're currently testing semgrep's diff-aware PR comment feature on a subset of our repos-- if you run into issues or find this spammy, please reach out to @danny.cooper in slack and give feedback. 

For backwards compatability with gosec, its best to use polyglot suppression comments of the following format for false positives:
`// #nosec <gosec rule ID> nosemgrep: <semgrep rule ID>`


<sub>*You can view more details about [this finding](https://semgrep.dev/orgs/grafana_labs/findings/297856441) in the Semgrep AppSec Platform.*</sub>

**Review by dehaansa — APPROVED** [review]
LGTM. Resolve the merge conflicts and we can get it merged!

**Comment by clayton-cornell** [comment]
@dehaansa was your comment about not tagging the docs as community resolved correctly? The docs still state that this is community supported.

**Comment by dehaansa** [comment]
> @dehaansa was your comment about not tagging the docs as community resolved correctly? The docs still state that this is community supported.

Ah, you're right. This doesn't need to be a community component.

File: docs/sources/reference/components/otelcol/otelcol.exporter.file.md [hunk 7]
```diff
@@ -0,0 +1,246 @@
+---
+canonical: https://grafana.com/docs/alloy/latest/reference/components/otelcol/otelcol.exporter.file/
+description: Learn about otelcol.exporter.file
+labels:
+  stage: public-preview
+  products:
+    - oss
+  tags:
+    - text: Community
+      tooltip: This component is developed, maintained, and supported by the Alloy user community.
```
Comment (clayton-cornell) [inline docs/sources/reference/components/otelcol/otelcol.exporter.file.md]:
```suggestion
    - oss
```

File: docs/sources/reference/components/otelcol/otelcol.exporter.file.md [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+title: otelcol.exporter.file
+---
+
+# `otelcol.exporter.file`
+
+{{< docs/shared lookup="stability/community.md" source="alloy" version="<ALLOY_VERSION>" >}}
+
+`otelcol.exporter.file` accepts metrics, logs, and traces telemetry data from other `otelcol` components and writes it to files on disk.
```
Comment (clayton-cornell) [inline docs/sources/reference/components/otelcol/otelcol.exporter.file.md:13]:
```suggestion
# `otelcol.exporter.file`

`otelcol.exporter.file` accepts metrics, logs, and traces telemetry data from other `otelcol` components and writes it to files on disk.
```

**Comment by madhub** [comment]
@dehaansa  can you help to resolve the  conflicts in CHANGELOG.md, go.mod, go.sum.

**Comment by clayton-cornell** [comment]
@madhub There's a couple conflicts that need resolution. Can you fix them up?

**Comment by dehaansa** [comment]
I've cleaned up the conflicts and fixed a couple bits - but the CLA assistant says that you haven't signed the CLA, can you re-sign it if it's mixed up (or sign it for the first time, if it's not!) https://github.com/grafana/alloy/pull/4475#issuecomment-3315140926

**Comment by dehaansa** [comment]
I've made this one a bit messy to try to sign past commits. Will need to be sure we clean up the merge commit when it's ready to merge.

Edit: I think you'll need to recreate from scratch probably to clean this up. Sorry for the inconvenience here, we added a requirement to sign commits and I had touched your branch with unsigned commits.

Edit2: We've backed off on the signed commits requirement, while this is a bit messy looking I think it'll be fine on merge. working on getting CI happy.
